### PR TITLE
Migrate to tf2

### PR DESCRIPTION
**Pull request by MatthijsBurgh**
Merge after https://github.com/tue-robotics/geolib2/pull/44

### CODE REVIEW

File: package.xml [hunk 1]
```diff
@@ -13,11 +18,21 @@
   <build_depend>ed</build_depend>
   <build_depend>geometry_msgs</build_depend>
   <build_depend>sensor_msgs</build_depend>
-  <build_depend>tf</build_depend>
+  <build_depend>tf2</build_depend>
+  <build_depend>tf2_geometry_msgs</build_depend>
```
Comment (jlunenburg) [inline package.xml:22]: Depend?

File: src/localization_tf_plugin.cpp [hunk 2]
```diff
@@ -34,32 +27,29 @@ LocalizationTFPlugin::~LocalizationTFPlugin()
 void LocalizationTFPlugin::configure(tue::Configuration config)
 {
     config.value("robot_name", robot_name_);
-
```
Comment (jlunenburg) [inline src/localization_tf_plugin.cpp:37]: Was there a specific reason here why the listener was deleted and restarted?

Comment (MatthijsBurgh) [reply src/localization_tf_plugin.cpp:37]: That did cross my mind too. But I couldn't think of a situation in which restarting would make a difference.

File: package.xml (hunk 1 above)
Comment (MatthijsBurgh) [reply package.xml:22]: `depend` is actually the lazy way.
`depend` means `build_depend`, `exec_depend` and `build_export_depend`. But none of these packages are actually included in exposed header files, as there aren't any. So therefore `depend` is not the most correct way to go.

**Review by jlunenburg — APPROVED** [review]
I assume this has been tested properly

**Pull request merged**